### PR TITLE
Fix additional warnings on Windows

### PR DESCRIPTION
**Pull request by sstamenova**
Technically, some of them would apply to other platforms as well. This brings down the warnings on Windows to just a few left.

Signed-off-by: Stella Stamenova <stilis@microsoft.com>

### CODE REVIEW

**Comment by sstamenova** [comment]
@sorenlassen : FYI

File: src/Transform/ONNX/ShapeInferencePass.cpp [hunk 1]
```diff
@@ -97,8 +97,9 @@ class ShapeInferencePass
   }
 
   static LogicalResult runShapeInferenceOnRegion(Region &r) {
-    std::function<void(Region &)> doShapeInference =
-        &ShapeInferencePass::runShapeInferenceOnRegion;
+    std::function<void(Region &)> doShapeInference = [&](Region &region) {
```
Comment (sorenlassen) [inline src/Transform/ONNX/ShapeInferencePass.cpp:100]: nit: can you omit the `&` in `[&]`? it doesn't seem like anything is captured

File: test/numerical/Common.hpp [hunk 2]
```diff
@@ -2,14 +2,14 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#if defined(__MSC_VER)
+#if defined(_MSC_VER)
```
Comment (sorenlassen) [inline test/numerical/Common.hpp:5]: thanks for fixing this!

**Review by sorenlassen — APPROVED** [review]
LGTM

File: src/Transform/ONNX/ShapeInferencePass.cpp (hunk 1 above)
Comment (AlexandreEichenberger) [reply src/Transform/ONNX/ShapeInferencePass.cpp:100]: if applicable, make sense to remove the captured context

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM after last comment from @sorenlassen is acted upon.

**Pull request merged**